snapshot/restore/local_fs_output.go

```diff
@@ -91,11 +91,11 @@ func (o *FilesystemOutput) Close(ctx context.Context) error {
 }
 
 // WriteFile implements restore.Output interface.
-func (o *FilesystemOutput) WriteFile(ctx context.Context, relativePath string, f fs.File) error {
+func (o *FilesystemOutput) WriteFile(ctx context.Context, relativePath string, f fs.File, ensureFileIsWrittenAtomic bool) error {
```

Comment:
instead of the flag you can use a field on FilesystemOutput itself. Just make sure the default value of false is reasonable (which it is in this case).